### PR TITLE
Add import endpoint

### PR DESCRIPTION
**Pull request by antekresic**
This new endpoint is supposed to be an easier way to ingest data using Promscale
by sending JSON streaming metric data.

### CODE REVIEW

File: pkg/api/write.go [hunk 1]
```diff
@@ -148,6 +158,96 @@ func validateWriteHeaders(w http.ResponseWriter, r *http.Request) bool {
...
+		numSamples, err := writer.Ingest(req.GetTimeseries(), req)
+		if err != nil {
+			log.Warn("msg", "Error importing samples to remote storage", "err", err, "num_samples", numSamples)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	})
+}
+
+func toWriteRequest(i importPayload, req *prompb.WriteRequest) *prompb.WriteRequest {
```
Comment (cevian) [inline pkg/api/write.go:220]:
```suggestion
func appendImportPayload(req *prompb.WriteRequest, i importPayload) *prompb.WriteRequest {
```

File: pkg/api/write.go [hunk 2]
```diff
@@ -148,6 +158,96 @@ func validateWriteHeaders(w http.ResponseWriter, r *http.Request) bool {
...
+	log.Error("msg", msg, "err", err)
 	http.Error(w, err, http.StatusBadRequest)
 }
+
+func buildImportError(w http.ResponseWriter, err string) {
+	buildError(w, err, "Import header validation error")
+}
+
+type importPayload struct {
+	Metric     map[string]string `json:"metric"`
```
Comment (JLockerman) [inline pkg/api/write.go:174]: is this to insert the same values to multiple metrics, or does it store the labels key-value mappings, or something else?

File: pkg/api/write.go [hunk 3]
```diff
@@ -148,6 +158,96 @@ func validateWriteHeaders(w http.ResponseWriter, r *http.Request) bool {
...
+				log.Error("msg", "Import error", "err", err.Error())
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			if len(i.Timestamps) != len(i.Values) {
+				log.Error("msg", "Import error", "err", "Mismatch number of timestamps and values in payload")
+				http.Error(w, "Mismatch number of timestamps and values in payload", http.StatusInternalServerError)
+				return
+			}
```
Comment (JLockerman) [inline pkg/api/write.go:206]: This feels like a bit of an awkward constraint to impose; is this better than having the input be an array of `(time, value)` pairs?

File: pkg/api/write.go (hunk 2 above)
Comment (JLockerman) [reply pkg/api/write.go:174]: Looking at the code, it seems to be the label key-value mappings. Can we name this `labels` or `series`?

**Review by JLockerman — COMMENTED** [review]
Over all it looks good, though I have some questions about the JSON format. If we do have some constraints forcing it to be the way it is, we should document them.

**Pull request closed without merging**